### ISSUE TITLE
Python tests fail to run on Windows

### ISSUE THREAD

**Issue by mchamberland** [clrp-code/egs_brachy#9]
This [line ](https://github.com/clrp-code/egs_brachy/blob/master/egs_brachy/run_tests.py#L123)fails on Windows because it assumes Unix-style path separators. It should use os.path.sep instead.

This [line ](https://github.com/clrp-code/egs_brachy/blob/master/egs_brachy/tests/iaea.py#L8)also fails because it is looking for a .so object, but it should be .dll on Windows. Also, on Windows, there is no "lib" prefix on the DLLs.

Working on a fix now.

**Comment by mchamberland** [comment]
For some reason, trying to remove the IAEA phsp files after running the phsp_scoring test results in a Permission Error:


```
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\m310767\\dev\\EGSnrc\\egs_home\\egs_brachy\\eb_test_run.phsp.IAEAheader'
```

It comes from this [line](https://github.com/clrp-code/egs_brachy/blob/master/egs_brachy/run_tests.py#L114). @randlet  can we force remove or simply skip if we get an exception?

**Comment by randlet** [comment]
Not sure there is a way to force deletion, but wrapping in a try/except to skip if there is an exception should be fine.

**Comment by mchamberland** [comment]
Sounds good. I have something like this now:

```
try:
  os.remove(f)
except PermissionError as e:
  print(e)
  print('{} was not deleted because of a permission error.'.format(os.path.basename(f)))
```

**Comment by SimonBiggs** [comment]
> This [line ](https://github.com/clrp-code/egs_brachy/blob/master/egs_brachy/run_tests.py#L123)fails on Windows because it assumes Unix-style path separators. It should use os.path.sep instead.
> 
> This [line ](https://github.com/clrp-code/egs_brachy/blob/master/egs_brachy/tests/iaea.py#L8)also fails because it is looking for a .so object, but it should be .dll on Windows. Also, on Windows, there is no "lib" prefix on the DLLs.
> 
> Working on a fix now.

I find Azure Pipelines are great for achieving testing on Windows Mac and Linux with minimal overhead. ... Can all be defined in the one Azure pipelines file...

**Comment by SimonBiggs** [comment]
If it helps, here is an example file https://github.com/pymedphys/pymedphys/blob/master/azure-pipelines.yml

**Comment by mchamberland** [comment]
Thanks, @SimonBiggs! Definitely something to look at for the future.

**Comment by mchamberland** [comment]
This issue should be fixed in 88e5887bb058c8324965ed96e544aead17678017.

**Comment by mchamberland** [comment]
Also, egs_brachy now works out of the box on Windows... at least, on my configuration!

I'm gonna try building egs_view to see if: 1) I can get it working properly with egs_glib and egs_autoenvelope and 2) know what people are in for when we ask them to compile it from source!
